concrete-core/docs/api/backends.md

```diff
@@ -0,0 +1,243 @@
++ [`LweCiphertextDecryptionEngine`](https://docs.rs/concrete-core/latest/concrete_core/specification/engines/trait.LweCiphertextDescryptionEngine.html)
++ and more
+
+If you read between the lines, the fact that we use traits to represent operators, means that we will have to
+use special object (which we call _engines_ in this document), to perform the operations. This is
+slightly different from an object model in which operators are usually tied to the data themselves. In
+practice, this makes sense because we tend to use _side-resources_ to perform operations, and those are better
+stored in a separate structure.
+
+Contrary to entities, there is no restrictions on how many `*Engine` traits a type can implement. In practice,
+we tend to expose very few engines in a given backend.
+
+## Example backend
+
+Let's go through an example of a backend implementation to ensure we understand the concepts clearly.
+
+The `core` backend, contains Zama's own cpu based implementation of the scheme. It is located at
+[`backends::core`](https://docs.rs/concrete-core/latest/concrete_core/backends/core/index.html). The
+associated feature flag is the `backend_core`, but since it is the most prominent backend for now, we include
+it automatically (it is part of the `default` flag).
+
+In this core backend, you will find the same structure as in the `specification`:
++ One [`engines`](https://docs.rs/concrete-core/latest/concrete_core/backends/core/engines/index.html) module
+  containing the engines exported by the `core` backend
++ One [`entities`](https://docs.rs/concrete-core/latest/concrete_core/backends/core/entities/index.html)
+  module containing the entities exported by the `core` backend
+
+In the `entities` module, among other types, we find the
+[`LweCiphertext64`](https://docs.rs/concrete-core/latest/concrete_core/backends/core/entities/struct.LweCiphertext64.html)
+type. It is an _entity_, which implements the
+[`LweCiphertextEntity`](https://docs.rs/concrete-core/latest/concrete_core/specification/entities/trait.LweCiphertextEntity.html)
+trait (this type is actually listed in the implementors of the type).
+
+In the `engines` module, we find a single type, which is
+[`CoreEngine`](https://docs.rs/concrete-core/latest/concrete_core/backends/core/engines/struct.CoreEngine.html)
+. It is an _engine_ which implements many `*Engine` traits, among which the
+[`LweCiphertextEncryptionEngine`](https://docs.rs/concrete-core/latest/concrete_core/specification/engines/trait.LweCiphertextEncryptionEngine.html)
+trait which it implements for 32 and 64 bits precisions.
+
+## Operator semantics
+
+As much as possible, we try to support different semantics for each operator in `specification::engines`.
+
+__Pure operators__ take their inputs as arguments, allocate the objects holding the results, and return them.
```

Comment:
maybe turn this paragraph into level3 sections with 
```
### Pure operators
### Discarding operators
### Fusing operators
```
to have them list in the table of content?